tests/it/src/commands/create_diff_cases.rs

```diff
@@ -0,0 +1,129 @@
+pub(super) mod function {
+    use anyhow::Context;
+    use std::{
+        collections::HashSet,
+        path::{Path, PathBuf},
+    };
+
+    use gix::{
+        bstr::{BString, ByteSlice},
+        objs::FindExt,
+    };
+
+    pub fn create_diff_cases(
+        sliders_file: PathBuf,
+        worktree_dir: &Path,
+        destination_dir: PathBuf,
+        count: usize,
+        asset_dir: Option<BString>,
+    ) -> anyhow::Result<()> {
+        // TODO: turn into parameter.
+        let dry_run = false;
+
+        let prefix = if dry_run { "WOULD" } else { "Will" };
+        let sliders = std::fs::read_to_string(&sliders_file)?;
+
+        eprintln!(
+            "read {} which has {} lines",
+            sliders_file.display(),
+            sliders.lines().count()
+        );
+
+        let sliders: HashSet<_> = sliders
+            .lines()
+            .take(count)
+            .map(|line| {
+                let parts: Vec<_> = line.split_ascii_whitespace().collect();
+
+                match parts[..] {
+                    [before, after, ..] => (before, after),
+                    _ => todo!(),
+                }
+            })
+            .collect();
+
+        eprintln!("{sliders:?}");
+
+        let repo = gix::open(worktree_dir)?;
+
+        let asset_dir = asset_dir.unwrap_or("assets".into());
+        let assets = destination_dir.join(asset_dir.to_os_str()?);
+
+        eprintln!("{prefix} create directory '{assets}'", assets = assets.display());
+        if !dry_run {
+            std::fs::create_dir_all(&assets)?;
+        }
+
+        let mut buf = Vec::new();
+
+        let script_name = "make_diff_for_sliders_repo.sh";
+
+        let mut blocks: Vec<String> = vec![format!(
+            r#"#!/usr/bin/env bash
+# TODO:
+# `git diff --no-index` returns 1 when there's differences, but 1 is treated as an error by the
```

Comment:
I'd keep the `set -eu…` and do `git diff … || true`  or something like that, assuming that it will keep working.